x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_creation/engine_creation.test.tsx

```diff
@@ -0,0 +1,118 @@
+
+import React from 'react';
+import { shallow } from 'enzyme';
+
+import { EngineCreation } from './';
+
+describe('EngineCreation', () => {
+  const DEFAULT_VALUES = {
+    name: '',
+    rawName: '',
+    language: 'Universal',
+  };
+
+  const MOCK_ACTIONS = {
+    setRawName: jest.fn(),
+    setLanguage: jest.fn(),
+    submitEngine: jest.fn(),
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+    setMockValues(DEFAULT_VALUES);
+    setMockActions(MOCK_ACTIONS);
+  });
+
+  it('renders', () => {
+    const wrapper = shallow(<EngineCreation />);
+    expect(wrapper.find('[data-test-subj="EngineCreation"]')).toHaveLength(1);
+  });
+
+  it('EngineCreationForm calls submitEngine on form submit', () => {
+    const wrapper = shallow(<EngineCreation />);
+    const simulatedEvent = {
+      preventDefault: jest.fn(),
+    };
+    wrapper.find('[data-test-subj="EngineCreationForm"]').simulate('submit', simulatedEvent);
+
+    expect(MOCK_ACTIONS.submitEngine).toHaveBeenCalledTimes(1);
+  });
+
+  it('EngineCreationNameInput calls setRawName on change', () => {
+    const wrapper = shallow(<EngineCreation />);
+    const simulatedEvent = {
+      currentTarget: { value: 'new-raw-name' },
+    };
+    wrapper.find('[data-test-subj="EngineCreationNameInput"]').simulate('change', simulatedEvent);
+
+    expect(MOCK_ACTIONS.setRawName).toHaveBeenCalledWith('new-raw-name');
+  });
+
+  it('EngineCreationLanguageInput calls setLanguage on change', () => {
+    const wrapper = shallow(<EngineCreation />);
+    const simulatedEvent = {
+      currentTarget: { value: 'English' },
+    };
+    wrapper
+      .find('[data-test-subj="EngineCreationLanguageInput"]')
+      .simulate('change', simulatedEvent);
+
+    expect(MOCK_ACTIONS.setLanguage).toHaveBeenCalledWith('English');
+  });
+
+  describe('NewEngineSubmitButton', () => {
+    it('is disabled when name is empty', () => {
+      setMockValues({ ...DEFAULT_VALUES, name: '', rawName: '' });
+      const wrapper = shallow(<EngineCreation />);
+
+      expect(wrapper.find('[data-test-subj="NewEngineSubmitButton"]').prop('disabled')).toEqual(
+        true
+      );
+    });
+
+    it('is enabled when name has a value', () => {
+      setMockValues({ ...DEFAULT_VALUES, name: 'test', rawName: 'test' });
+      const wrapper = shallow(<EngineCreation />);
+
+      expect(wrapper.find('[data-test-subj="NewEngineSubmitButton"]').prop('disabled')).toEqual(
+        false
+      );
+    });
+  });
+
+  describe('EngineCreationNameFormRow', () => {
+    it('renders sanitized name helptext when the raw name is being sanitized', () => {
+      setMockValues({
+        ...DEFAULT_VALUES,
+        name: 'un-sanitized-name',
+        rawName: 'un-----sanitized-------name',
+      });
+      const wrapper = shallow(<EngineCreation />);
+      const formRow = wrapper.find('[data-test-subj="EngineCreationNameFormRow"]').dive();
+
+      expect(formRow.contains('Your engine will be named')).toBeTruthy();
```

Comment:
This assertion feels like it's missing the actual name? Is there any way we can assert on that?

```suggestion
      expect(formRow.contains('Your engine will be named un-sanitized-name')).toBeTruthy();
```

not sure if the DOM works out that way, but that would make the test make a lot more sense when reading it. If it takes more than 5 minutes to do or it's a pain than no worries.